cmd/object-handlers.go

```diff
@@ -547,19 +699,51 @@ func (api objectAPIHandlers) GetObjectHandler(w http.ResponseWriter, r *http.Req
 	})
 }
 
-// HeadObjectHandler - HEAD Object
-// -----------
-// The HEAD operation retrieves metadata from an object without returning the object itself.
-func (api objectAPIHandlers) HeadObjectHandler(w http.ResponseWriter, r *http.Request) {
-	ctx := newContext(r, w, "HeadObject")
+// getFilesFromZIPObject reads a partial stream of a zip file to build the zipindex.Files index
+func getFilesListFromZIPObject(ctx context.Context, objectAPI ObjectLayer, bucket, object string) (zipindex.Files, ObjectInfo, error) {
+	var size = 10 << 10
```

Comment:
```suggestion
	var size = 1 << 20
```
Much more likely we will have the full directory with this. Cost is close to the same.